extensions/metrics/src/metrics.component.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/extension-metrics
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {createBindingFromClass, inject} from '@loopback/context';
+import {Application, Component, CoreBindings} from '@loopback/core';
+import {metricsControllerFactory} from './controllers';
+import {MetricsInterceptor} from './interceptors';
+import {MetricsBindings} from './keys';
+import {MetricsObserver, PushGatewayObserver} from './observers';
+import {DEFAULT_METRICS_OPTIONS, MetricsOptions} from './types';
+
+/**
+ * A component providing metrics for Prometheus
+ */
+export class MetricsComponent implements Component {
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE)
+    private application: Application,
+    @inject(MetricsBindings.CONFIG, {optional: true})
+    options: MetricsOptions = DEFAULT_METRICS_OPTIONS,
```

Comment:
Should we use a `@config`-based binding key here? I find it confusing that certain parts of the app are configured via `app.configure` and other parts via `app.bind`.

```
    @config()
    options: MetricsOptions = DEFAULT_METRICS_OPTIONS,
```